core/pkg/ingress/controller/controller.go

```diff
@@ -1068,24 +1032,54 @@ func (ic *GenericController) serviceEndpoints(svcKey, backendPort string,
 	return upstreams, nil
 }
 
+// createUpstreamEndpoint returns the upstream servers (endpoints) associated
+// to a ingress.
+func (ic *GenericController) createUpstreamEndpoint(
+	upstreamName, svcKey string,
+	serviceUpstream bool,
+	backend *extensions.IngressBackend,
+	upstreams map[string]*ingress.Backend,
+	hz *healthcheck.Upstream,
+) map[string]*ingress.Backend {
+	// Add the service cluster endpoint as the upstream instead of individual endpoints
+	// if the serviceUpstream annotation is enabled
+	if serviceUpstream {
+		endpoint, err := ic.getServiceClusterEndpoint(svcKey, backend)
+		if err != nil {
+			glog.Errorf("Failed to get service cluster endpoint for service %s: %v", svcKey, err)
+		} else {
+			upstreams[upstreamName].Endpoints = []ingress.Endpoint{endpoint}
+		}
+	}
+
+	if len(upstreams[upstreamName].Endpoints) == 0 {
+		endps, err := ic.serviceEndpoints(svcKey, backend.ServicePort.String(), hz)
```

Comment:
previously here panic. Now I pass extensions.IngressBackend as a param to cover 2 cases